src/functions.php

```diff
@@ -77,7 +77,7 @@ function uri_for($uri)
 function stream_for($resource = '', array $options = [])
 {
     switch (gettype($resource)) {
-        case 'string':
+        case is_scalar($resource):
```

Comment:
I don't think it makes sense to use `stream_for()` with a boolean value, but it's probably better than add two `case` statements (`float` and `int`) to do this validation, it's too much complexity for not so strong reasons.